### PR TITLE
Fixed issue with selecting namespace as all in picklist, and where de…

### PR DESCRIPTION
**Pull request by kevinylee1991**
…fault namespace from user preference was being ignored

### CODE REVIEW

File: src/app/core/components/grid/useGridFiltering.ts [hunk 1]
```diff
@@ -237,7 +238,7 @@ export default function useGridFiltering<
           valuesByKey[columnKey] !== undefined &&
           (allowEmpty || !isNilOrEmpty(valuesByKey[columnKey]))
         ) {
-          if (controlled) {
+          if (controlled || valuesByKey[columnKey] === allKey) {
```
Comment (kevinylee1991) [inline src/app/core/components/grid/useGridFiltering.ts:241]: without this, selecting All would make it filter into an empty list

File: src/app/plugins/kubernetes/components/common/NamespacePicklist.tsx [hunk 2]
```diff
@@ -64,19 +67,12 @@ export default function NamespacePicklist({
   useEffect(() => {
     if (!clusterId) return
     // Reset the namespace when the clusterId changes
-    onChange(allKey)
+    onChange('')
```
Comment (kevinylee1991) [inline src/app/plugins/kubernetes/components/common/NamespacePicklist.tsx:70]: I think we do not want to default to showing all namespaces right?

File: src/app/plugins/kubernetes/components/common/NamespacePicklist.tsx [hunk 3]
```diff
@@ -64,19 +67,12 @@ export default function NamespacePicklist({
...
   }, [clusterId])
 
   // Select the first item as soon as data is loaded
   useEffect(() => {
     if (!clusterId || value) return
-
```
Comment (kevinylee1991) [inline src/app/plugins/kubernetes/components/common/NamespacePicklist.tsx:73]: This part seems to be unnecessary with the new Dropdown component

File: src/app/plugins/kubernetes/components/rbac/profiles/create/roles-form.tsx [hunk 4]
```diff
@@ -133,6 +133,8 @@ const RolesForm = ({ wizardContext, setWizardContext }) => {
       >
         <RolesListTable
           filters={
+            // fixme: This namespace picklist does not work with clusterId as allkey due to
+            // validateParams method in ListAction. Until fixed this will only have all option
             <NamespacePicklist
```
Comment (kevinylee1991) [inline src/app/plugins/kubernetes/components/rbac/profiles/create/roles-form.tsx:138]: Low priority fix i think, can do it later

**Review by jruddell — APPROVED** [review]

**Pull request merged**